subs/pantry/src/Pantry/Storage.hs

```diff
@@ -134,8 +140,17 @@ HackageCabal
     -- If available: the full tree containing the HackageTarball
     -- contents with the cabal file modified.
     tree TreeId Maybe
+    -- If available should be equal to True meaning that
+    -- exposed modules were saved in HackageExposedModules
+    hasExposedModules Bool Maybe
     UniqueHackage name version revision
 
+-- Cache of modules exposed by a Hackage package
+HackageExposedModules
+    cabal BlobId
+    moduleName P.ModuleNameP
```

Comment:
Since there are likely to be a lot of duplicated module names, this would probably benefit from normalization, the same way we normalize package names. How about:

```
    moduleName ModuleId

Module
    name P.ModuleName
    UniqueModule name
```

Or something like that?